src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -265,24 +263,20 @@ public static void run(Settings settings, MessageRouter<MessageGroupBatch> messa
                         } else {
                             fixMessage = MessageUtils.parse(session, strMessage);
                         }
-
-                        if (!session.send(fixMessage)) {
-                            LOGGER.error("Message not sent. Message was not queued for transmission to the counterparty");
-
-                            EventID eventID = message.getMessage().getParentEventId();
-                            String parentEventID = eventID.getId().isEmpty() ? rootEventID : eventID.getId();
-                            MessageID messageID = message.getMessage().getMetadata().getId();
-                            Event event = Event.start().messageID(messageID).type("Error").status(Event.Status.FAILED);
-                            MessageRouterUtils.storeEvent(eventRouter, event, parentEventID);
+                        FixMessage fixMessage1 = new FixMessage(fixMessage, message.getRawMessage().getParentEventId());
```

Comment:
> new FixMessage

We can avoid its creation altogether if message has no parent-event-id

> fixMessage1

Seriously?